teku/src/main/java/tech/pegasys/teku/cli/util/DatabaseMigrater.java

```diff
@@ -0,0 +1,256 @@
+      Files.move(
+          dataDirLayout.getBeaconDataDirectory(),
+          getMovedOldBeaconFolderPath(),
+          StandardCopyOption.ATOMIC_MOVE);
+    } catch (IOException ex) {
+      statusUpdater.accept(ex.getMessage());
+      throw new DatabaseMigraterError(
+          "Failed to move old database to " + getMovedOldBeaconFolderPath().toString());
+    }
+    try {
+      Files.move(
+          getNewBeaconFolderPath(),
+          dataDirLayout.getBeaconDataDirectory(),
+          StandardCopyOption.ATOMIC_MOVE);
+    } catch (IOException ex) {
+      statusUpdater.accept(ex.getMessage());
+      throw new DatabaseMigraterError(
+          "Failed to move new database to " + dataDirLayout.getBeaconDataDirectory().toString());
+    }
+  }
+
+  @VisibleForTesting
+  void closeDatabases() {
+    try {
+      originalDatabase.close();
+    } catch (Exception e) {
+      statusUpdater.accept("Failed to close original database cleanly: " + e.getMessage());
+    }
+    try {
+      newDatabase.close();
+    } catch (Exception e) {
+      statusUpdater.accept("Failed to close new database cleanly: " + e.getMessage());
+    }
+  }
+
+  @VisibleForTesting
+  void duplicateBeaconFolderContents() throws IOException {
+    final Path newBeaconFolderPath = getNewBeaconFolderPath();
+    if (newBeaconFolderPath.toFile().isDirectory()) {
+      FileUtils.deleteDirectory(newBeaconFolderPath.toFile());
+    }
+    newBeaconFolderPath.toFile().mkdir();
+    for (String currentEntry : List.of("network.yml", "kvstore")) {
+      Path currentPath = dataDirLayout.getBeaconDataDirectory().resolve(currentEntry);
+      if (Files.exists(currentPath)) {
+        if (currentPath.toFile().isDirectory()) {
+          FileUtils.copyDirectory(
+              currentPath.toFile(), newBeaconFolderPath.resolve(currentEntry).toFile());
+        } else {
+          Files.copy(currentPath, newBeaconFolderPath.resolve(currentEntry));
+        }
+      }
+    }
+    migratedDatabasePath = Optional.of(newBeaconFolderPath);
+  }
+
+  @VisibleForTesting
+  void migrateData() throws DatabaseMigraterError {
+    try {
+      newDatabase.ingestDatabase(originalDatabase, batchSize, statusUpdater);
+    } catch (Exception ex) {
+      throw new DatabaseMigraterError(
+          "Failed to migrate data into the new database: " + ex.getCause());
+    }
+  }
+
+  @VisibleForTesting
+  KvStoreDatabase createDatabase(final Path databasePath, DatabaseVersion databaseVersion) {
+    final VersionedDatabaseFactory databaseFactory =
+        new VersionedDatabaseFactory(
+            new NoOpMetricsSystem(),
+            databasePath,
+            dataStorageOptions.getDataStorageMode(),
+            databaseVersion,
+            DEFAULT_STORAGE_FREQUENCY,
+            eth2NetworkOptions.getNetworkConfiguration().getEth1DepositContractAddress(),
+            dataStorageOptions.isStoreNonCanonicalBlocks(),
+            spec);
+    return (KvStoreDatabase) databaseFactory.createDatabase();
```

Comment:
Might be worth checking this is a `KvStoreDatabase` and returning a more helpful error message about it being an unsupported target format if not.